linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -5660,7 +5671,10 @@ private[optimizer] object OptimizerCore {
   private val ClassTagApplyMethodName =
     MethodName("apply", List(ClassRef(ClassClass)), ClassRef(ClassName("scala.reflect.ClassTag")))
 
-  final class InlineableClassStructure(private val allFields: List[FieldDef]) {
+  final class InlineableClassStructure(val className: ClassName, private val allFields: List[FieldDef]) {
+    private[OptimizerCore] val refinedType: RefinedType =
```

Comment:
What is this useful for? IIUC this is only used in `tpe` of `PreTransRecordTree`. And that usage feels very odd (I have commented there).